### PR TITLE
required keyholders to be vetted for validity

### PR DESCRIPTION
**Pull request by laftho**

### CODE REVIEW

File: app/domain/User.php [hunk 1]
```diff
@@ -127,6 +127,13 @@ public function get_valid() {
         $privs = $this->getPrivilegeCodes();
         if( in_array( "administrator", $privs ) )
             return true;
+
+        $keyholder = Membership::findByCode(Membership::KEYHOLDER)[0];
+
+        // non-vetted but active keyholder
+        if ($this->membership_id == $keyholder->id && !in_array("vetted", $privs)) {
+          return false;
```
Comment (TyIsI) [inline app/domain/User.php]: Isn't this supposed to be true?

Comment (TyIsI) [reply app/domain/User.php]: This is to ensure users are not seen as valid.

Comment (laftho) [reply app/domain/User.php]: It should be false when they aren't vetted

File: app/services/UserService.php [hunk 2]
```diff
@@ -260,13 +260,6 @@ public function UpdateMembership($userid, $membershipid) {
             throw new \Exception("Invalid user or membership type");
         }
 
-        if ($membership->code == Membership::KEYHOLDER && !$user->privileges->contains(Privilege::findByCode("vetted"))) {
```
Comment (laftho) [inline app/services/UserService.php]: It's probably ok to do this with the vetted check in valid because we'll see more of these as people upgrade

**Pull request merged**